src/TimeSteppers/clock.jl

```diff
@@ -6,30 +6,31 @@ import Base: show
 import Oceananigans.Units: Time
 
 """
-    mutable struct Clock{T<:Number}
+    mutable struct Clock{T, FT}
 
-Keeps track of the current `time`, `iteration` number, and time-stepping `stage`.
+Keeps track of the current `time`, `last_Δt`, `iteration` number, and time-stepping `stage`.
 The `stage` is updated only for multi-stage time-stepping methods. The `time::T` is
 either a number or a `DateTime` object.
 """
-mutable struct Clock{T}
+mutable struct Clock{T, FT}
          time :: T
+      last_Δt :: FT

```

Comment:
```suggestion
      last_Δt :: DT
```

This will stand the test of time, when we support date times and time-deltas for the time-step.